src/web_peer_connection.cpp

```diff
@@ -789,9 +789,11 @@ void web_peer_connection::on_receive(error_code const& error
 			// if the status code is not one of the accepted ones, abort
 			if (!is_ok_status(m_parser.status_code()))
 			{
-				file_request_t const& file_req = m_file_requests.front();
-				m_web->have_files.resize(t->torrent_file().num_files(), true);
-				m_web->have_files.clear_bit(file_req.file_index);
+				if (!m_file_requests.empty())
+				{
+					file_request_t const& file_req = m_file_requests.front();
+					m_web->have_files.clear_bit(file_req.file_index);
```

Comment:
hm.. I didn't realize this was the previous behavior. including the resize(.., true); I apologize, I remember what the semantics of this field it. It does make sense that the fundamental assumption is that the web server has every file, and only when we fail to get a file from it do we mark this as false. I still think it would be a bit more conservative to look for specific error codes, but I don't expect your patch to do that.

Would you mind restoring this. Your patch was just to protect this against an empty ``m_file_requests``, right?